src/lib/util/options.cpp

```diff
@@ -692,252 +890,143 @@ std::string core_options::output_help() const
 //-------------------------------------------------
 
-void core_options::remove_entry(core_options::entry &delentry)
+bool core_options::header_exists(const char *description) const
 {
-	// remove all names from the map
-	for (int name = 0; name < ARRAY_LENGTH(delentry.m_name); name++)
-		if (!delentry.m_name[name].empty())
-		{
-			auto entry = m_entrymap.find(delentry.m_name[name]);
-			if (entry!= m_entrymap.end()) m_entrymap.erase(entry);
-		}
-
-	// remove the entry from the list
-	m_entrylist.remove(delentry);
-}
-
-/**
- * @fn  void core_options::copyfrom(const core_options &src)
- *
- * @brief   -------------------------------------------------
- *            copyfrom - copy options from another set
- *          -------------------------------------------------.
- *
- * @param   src Source for the.
- */
-
-void core_options::copyfrom(const core_options &src)
-{
-	// reset ourselves first
-	reset();
-
-	// iterate through the src options and make our own
-	for (entry &curentry : src.m_entrylist)
-		append_entry(*global_alloc(entry(curentry.name(), curentry.description(), curentry.flags(), curentry.default_value())));
-}
-
-/**
- * @fn  bool core_options::validate_and_set_data(core_options::entry &curentry, const char *newdata, int priority, std::string &error_string)
- *
- * @brief   -------------------------------------------------
- *            validate_and_set_data - make sure the data is of the appropriate type and within
- *            range, then set it
- *          -------------------------------------------------.
- *
- * @param [in,out]  curentry        The curentry.
- * @param   newdata                 The newdata.
- * @param   priority                The priority.
- * @param [in,out]  error_string    The error string.
- *
- * @return  true if it succeeds, false if it fails.
- */
-
-bool core_options::validate_and_set_data(core_options::entry &curentry, std::string &&data, int priority, std::string &error_string)
-{
-	// let derived classes override how we set this data
-	if (override_set_value(curentry.name(), data))
-		return true;
-
-	// validate the type of data and optionally the range
-	float fval;
-	int ival;
-	switch (curentry.type())
-	{
-	// booleans must be 0 or 1
-	case OPTION_BOOLEAN:
-		if (sscanf(data.c_str(), "%d", &ival) != 1 || ival < 0 || ival > 1)
-		{
-			error_string.append(string_format("Illegal boolean value for %s: \"%s\"; reverting to %s\n", curentry.name(), data.c_str(), curentry.value()));
-			return false;
-		}
-		break;
-
-	// integers must be integral
-	case OPTION_INTEGER:
-		if (sscanf(data.c_str(), "%d", &ival) != 1)
-		{
-			error_string.append(string_format("Illegal integer value for %s: \"%s\"; reverting to %s\n", curentry.name(), data.c_str(), curentry.value()));
-			return false;
-		}
-		if (curentry.has_range() && (ival < atoi(curentry.minimum()) || ival > atoi(curentry.maximum())))
+	auto iter = std::find_if(
+		m_entries.begin(),
+		m_entries.end(),
+		[description](const auto &entry)
```

Comment:
This is just one example, but please use two levels of indent for statement continuation, so it's obviously different to a new scope level.

```
    if (foo)
    {
        // One additional indent for new level of scope
        bar(
                arg1, // two additional indents for statement continuation
                arg2);
    }
```